service/matching/taskQueueManager.go

```diff
@@ -214,13 +206,18 @@ func newTaskQueueManager(
 	return tlMgr, nil
 }
 
-func (c *taskQueueManagerImpl) errShouldUnload(err error) bool {
-	return err != nil && (errIndicatesForeignLessee(err) || !c.config.ResilientSyncMatch())
-}
-
-func errIndicatesForeignLessee(err error) bool {
-	var condfail *persistence.ConditionFailedError
-	return errors.As(err, &condfail)
+// signalIfFatal calls signalFatalProblem on this taskQueueManagerImpl instance
+// if and only if the supplied error represents a fatal condition, e.g. the
+// existence of another taskQueueManager newer lease. Returns true if the signal
+// is emitted, false otherwise.
+func (c *taskQueueManagerImpl) signalIfFatal(err error) bool {
+	if err != nil && (errIndicatesForeignLessee(err) || !c.config.ResilientSyncMatch()) {
+		c.logger.Info("Task queue manager signaling a fatal error", tag.Error(err))
+		c.metricScope().IncCounter(metrics.ConditionFailedErrorPerTaskQueueCounter)
```

Comment:
Dumb mistake. Can't increment this counter here because we might be in this block just because ResilientSyncMatch is off.

Worth noting: the concrete error type probably should have been a label on a more generic error counter. Otherwise we churn our counter names if we change errors.